src/p2p_async.rs

```diff
@@ -43,13 +37,17 @@ impl<T: AsyncRead + AsyncWrite + Unpin + Send> P2psConnAsync<T> {
         stream.write_all(&keys.get_public_key_bytes()).await?;
 
         // create encryption key with private key and their public key
-        let key = keys.generate_encryption_key(&Keys::public_key_from_bytes(buffer)?);
+        let key = keys.generate_encryption_key(&Keys::public_key_from_bytes(buffer)?)?;
```

Comment:
Since no error is going to be thrown from **public_key_from_bytes**, there is no need to expect an error
```suggestion
        let key = keys.generate_encryption_key(&Keys::public_key_from_bytes(buffer))?;
```